nsis/src/nsis/file/NsisExecutable.java

```diff
@@ -45,13 +52,13 @@ private void findHeaderOffset() {
 			}
 		}
 	}
-	
+
 	public long getHeaderOffset() {
 		return this.header_offset;
 	}
-	
+
 	public long getFileLength() {
-		if(reader != null) {
+		if (reader != null) {
 			try {
 				return reader.length();
 			} catch (IOException e) {
```

Comment:
Do not catch the exception. If things are so broken that we can't get the size of the file we wan the exception to happen. Also, if you are calling this function and reader is null things are equally hopefully so we should probably thrown an exception.

I hope we can get rid of this function, see the other comment.

Reply:
The error handling changed since I deleted the function getFileLength(). The error is catched in findHeaderOffset, so if the file length is 0 the header offset is set to -1. That way, no error is thrown when a non NSIS file is opened while the NSIS plugin is active in Ghidra. I think that makes sense

Reply:
Changed